### PR TITLE
Cross-compile for windows without errors and opencl support

### PR DESCRIPTION
**Pull request by easdue**
Fixed compiler warnings end errors while cross-compiling with mingw64
Added support for opencl on windows. Compiled kernels are stored as <name>.bin and md5 sum is stored as <name>.md5

### CODE REVIEW

**Comment by TurboGit** [comment]
Nice to see this project moving forward! One general comment, what are you supporting? I mean what are the Gtk libraries used? Do you support gphoto? As darktable is uslng lot of external libraries it would be nice to know what is the current scope for this patch.

**Comment by easdue** [comment]
Ok, here goes:

gtk+3.18.6
gdk-pixbuf-2.32.3
glib-2.46.2
pango-1.38.1
atk-2.18.0
libxml2.9.2
libsoup-2.52.2
libwebp0.4.4
lensfun-0.3.2.0
gio-2.46.2
libgthread-2.0
libgmodule-2.0
libpangcairo-1.0
libwinpthread-4.0.4
librsvg-2.40.12
libsqlite-3.8.4.3
libcurl-7.42.0
libz-1.2.8
libpng-1.6.19
libjpeg-turbo-1.3.1
libtiff-4.0.3
liblcms-2.7.2
json-glib-1.0.4
exiv2-0.25
libiconv-0.0.6
libflickcurl-1.26
libopenjpeg1-1.5.2
libsecret-0.18.3
GraphicsMagick-1.3.23
libSDL-1.2.15

CUPS is not going to work i guess, will have to come up with something else
Compiling with gphoto2 added failes with a mysterious error that I haven't been able to figure out yet will come back to that.

**Comment by LebedevRI** [comment]
Ok, bad karma: here i come :)

Tl;dr: join irc.freenode.net, channel #darktable and talk to us.

http://article.gmane.org/gmane.comp.graphics.darktable.devel/6920/

```
So, what exactly do we mean by a maintainer?
0. Be actively involved with darktable for some time.
< CURRENT SITUATION: pr #1086, well, not nope. >

At least half a year.
< CURRENT SITUATION: nope >

That means:
*) Stick with us in IRC.
< CURRENT SITUATION: *REALLY BIG NOPE* - really important point. >

*) actually know the code.
*) actively examine bugreports, fix bugs.
1. Be reliable and trustworthy [over that time], and more importantly,
make us believe in that.
```

Guys, just force-feeding some random diff into us will not work.
Even https://github.com/darktable-org/darktable/blob/master/CONTRIBUTING.md says that you need to talk to us :(

As for $indows-related changes: my mail was 100% correct: until those conditions are met, NOTHING will happen.
And that includes even the changes that are needed to make dt compile on that [currently unsupported] platform.
Because they are the changes to help building on unsupported platform.
